sdk/eventgrid/azure-eventgrid/azure/eventgrid/_base_client.py

```diff
@@ -0,0 +1,59 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import (  # pylint: disable=unused-import
+    Any,
+    List,
+    Union,
+    TYPE_CHECKING
+)
+
+from azure.core.pipeline.policies import (
+    RequestIdPolicy,
+    HeadersPolicy,
+    RedirectPolicy,
+    RetryPolicy,
+    ContentDecodePolicy,
+    CustomHookPolicy,
+    NetworkTraceLoggingPolicy,
+    ProxyPolicy,
+    DistributedTracingPolicy,
+    HttpLoggingPolicy,
+    UserAgentPolicy
+)
+from ._helpers import _get_authentication_policy
+from ._policies import CloudEventDistributedTracingPolicy
+
+if TYPE_CHECKING:
+    from azure.core.credentials import AzureKeyCredential
+    from ._shared_access_signature_credential import EventGridSharedAccessSignatureCredential
+
+
+class PublisherClientMixin(object):  # pylint: disable=too-many-instance-attributes
```

Comment:
The mixin is overkill here, since no code is shared between sync and asyncm this just make the structure more complicated for no value

Reply:
I will change these to helper methods